azurelinuxagent/ga/firewall_manager.py

```diff
@@ -138,88 +157,98 @@ def _get_state_command(self):
         raise NotImplementedError()
 
 
-class _FirewallManagerMultipleRules(FirewallManager):
+class _FirewallManagerIndividualRules(FirewallManager):
     """
-    Base class for firewall managers that handle multiple rules, each rule being manipulated with a different command line (e.g. iptables, firewalld)
+    Base class for firewall managers (iptables, firewalld) that manipulate the firewall rules individually when checking/adding/removing them. For contrast, nft manipulated the entire table.
     """
-    def setup(self):
-        for command in self._get_commands(self._get_append_command_option()):
-            shellutil.run_command(command[1])
+    def __init__(self, wire_server_address):
+        super(_FirewallManagerIndividualRules, self).__init__(wire_server_address)
+        #
+        # We use this array to iterate over all the firewall rules when setting/checking/removing them.
+        # The order of the items is critical since we process each item sequentially and the firewall rules will follow the order in the array: the first
+        # item will be at the top of the chain, etc.
+        #
+        # Each item in the array is a tuple with the friendly name of the rule and a function that returns the command used to process that rule. This function
+        # takes as argument the option that is passed to the corresponding command  (-A, -C, and -D for iptables, and --passthrough, --query-passthrough, and --remove-passthrough
+        # for firewallcmd)
+        #
+        self._firewall_commands = [
+            (FirewallManager.ACCEPT_DNS, self._get_accept_dns_rule_command),
+            (FirewallManager.ACCEPT, self._get_accept_rule_command),
+            (FirewallManager.DROP, self._get_drop_rule_command)
+        ]
```

Comment:
This list used to be returned by the __get_commands() generator method (now deleted). Using an array allows more flexibility.